prov/gni/src/gnix_datagram.c

```diff
@@ -393,9 +409,10 @@ int gnix_dgram_hndl_free(struct gnix_dgram_hndl *the_hndl)
 		gnix_list_del_init(&dg_ptr->list);
 	}
 
-	list_for_each_safe(&the_hndl->wc_datagram_active_list,
+	list_for_each_safe(&the_hndl->wc_dgram_active_list,
 				p, next, list) {
 		dg_ptr = p;
+		if (dg_ptr->state == GNIX_DGRAM_STATE_FREE) continue;
```

Comment:
I'd prefer to see this: 

```
list_for_each_safe(&the_hndl->bnd_dgram_active_list, p, next, list) {
	dg_ptr = p;
	if (dg_ptr->state == GNIX_DGRAM_STATE_FREE) continue;
	status = GNI_EpPostDataCancel(dg_ptr->gni_ep);
	if ((status != GNI_RC_SUCCESS) &&
		(status != GNI_RC_NO_MATCH)) {
		ret = gnixu_to_fi_errno(status);
		goto err;
	}
	gnix_list_del_init(&dg_ptr->list);
}
```

Expressed like this: 
```
list_for_each_safe(&the_hndl->bnd_dgram_active_list, p, next, list) {
	dg_ptr = p;
	if (dg_ptr->state != GNIX_DGRAM_STATE_FREE) {
		status = GNI_EpPostDataCancel(dg_ptr->gni_ep);
		if ((status != GNI_RC_SUCCESS) &&
			(status != GNI_RC_NO_MATCH)) {
			ret = gnixu_to_fi_errno(status);
			goto err;
		}
		gnix_list_del_init(&dg_ptr->list);
	}
}
```
I believe they should be the same, but I think the second is more readable. At first I didn't notice the continue being the on the same line and it was somewhat confusing. It should be just as efficient as well.